### ISSUE TITLE
Swagger editor allows to define object with required attributes and no definition

### ISSUE THREAD

**Issue by kejmyn** [swagger-api/swagger-editor#4669]
### Q&A (please complete the following information)
 - OS: macOS
 - Browser: Firefox
 - Version: 120.0.1
 - Swagger-Editor version: latest: https://editor.swagger.io/
 - Swagger/OpenAPI version: 3.0.1

### Content & configuration
Using the online demo at editor.swagger.io.

Example Swagger/OpenAPI definition:
```yaml
 schema:
            type: object
            required:
              - param1
              - param2
            properties:
              param1:
                type: string
```

### Describe the bug you're encountering
The editor does not detect such definition as an error, there is no checking if attributes defined as required are defined in "properties".

**Comment by frankkilcommins** [comment]
Hi @kejmyn,

What you have defined is valid JSON Schema hence no error is rendered within the editor. 

By default, JSON Schema has `additionalProperties` set to true. That means any JSON instance to be validated against the schema can have properties whose names are not listed in the `properties` keyword.

With the schema you have defined, the following JSON instance is valid:
```
{
  "param1": "hello",
  "param2": "world",
  "foo": "bar",
  "more": "stuff"
}
```

If you want to restrict your JSON Schema so that all properties must be defined in `properties` keyword, then you should set `additionalProperties: false`.

That being said as JSON Schema is a constraints based format. One can think of a JSON Schema as just a collection of assertions. As a result it can be possible to define assertions that a mutually exclusive and can cause issues when running validations.

FYI - we've improved the schema linting in the _next_ version of our editor - https://editor-next.swagger.io/

There if you defined a schema with mutually exclusive assertions, like:
```
  SomeObject:        
    type: object
    required:
      - param1
      - param2
    properties:
      param1:
        type: string  
    additionalProperties: false
```
You will get a schema linting error like:
![image](https://github.com/swagger-api/swagger-editor/assets/15875424/6772481c-b57c-40d8-a805-7037a55e3eb4)

So in summary, what you reported is not an error but there are situations like what I described in the `SomeObject` schema above which should be flagged. These situations are not flagged in the current editor but they are flagged in the **next** editor version. We won't be fixing these in the current editor as it will soon be replaced by it's successor
